riscv/src/compiler.rs

```diff
@@ -444,6 +456,30 @@ fn preamble() -> String {
     reg Y[<=];
     reg Z[<=];
     reg W[<=];
+    reg A0[<=];
+    reg A1[<=];
+    reg A2[<=];
+    reg A3[<=];
+    reg A4[<=];
+    reg A5[<=];
+    reg A6[<=];
+    reg A7[<=];
+    reg A8[<=];
+    reg A9[<=];
+    reg A10[<=];
+    reg A11[<=];
+    reg P0;
+    reg P1;
+    reg P2;
+    reg P3;
+    reg P4;
+    reg P5;
+    reg P6;
+    reg P7;
+    reg C0;
+    reg C1;
+    reg C2;
+    reg C3;
```

Comment:
Yes this is not ideal. We need 12 extra state registers to store the data that goes into the machine, and 12 extra assignment registers for the instruction. If we find a way to do this in a lower level we could save columns.